Source/ACE/Entity/Player.cs

```diff
@@ -1577,6 +1578,73 @@ private void HandleActionTeleToMarketplaceInternal()
             DoMotion(motionMarketplaceRecall);
         }
 
+        public void HandleActionFinishBarber(ClientMessage message)
+        {
+            // Read the payload sent from the client...
+            PaletteGuid = message.Payload.ReadUInt32();
+            Character.HeadObject = message.Payload.ReadUInt32();
```

Comment:
These operations modify the Player (through the character).  Therefore they need to be run as part of the Player's Action's.  Do this through an ActionChain (just put the entire thing inside an ActionChain).